sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -261,29 +262,43 @@ def _check_interaction_cst(self, n_features):
         if self.interaction_cst is None:
             return None
 
-        if not (
-            isinstance(self.interaction_cst, Iterable)
-            and all(isinstance(x, Iterable) for x in self.interaction_cst)
-        ):
-            raise ValueError(
-                "Interaction constraints must be None or an iterable of iterables, "
-                f"got: {self.interaction_cst!r}."
-            )
+        if isinstance(self.interaction_cst, str):
+            if self.interaction_cst == "no interactions":
```

Comment:
I don't think we have any string hyper-parameter values in scikit-learn with a white space in them. Intuitively I would use replace the whitespace with an underscore but I did not do a consistency review of current hyper-parameter values in the code base.

It might be possible to conduct such a review programmatically thanks to `sklearn.utils.all_estimator` and the `_parameter_constraints` attributes of each estimator class.

Reply:
Ah, I have no strong feelings on this. I used a whitespace because I thought it is "by humans, for humans" - it just looks prettier to me like this.

```python
In [15]: for estimator_name, klass in sklearn.utils.all_estimators():
    ...:     if not hasattr(klass, "_parameter_constraints"):
    ...:         continue
    ...:     for name, constraints in klass._parameter_constraints.items():
    ...:         for constraint in constraints:
    ...:             if isinstance(constraint, sklearn.utils._param_validation.StrOptions):
    ...:                 print(estimator_name, name, constraint.options)
```
says you are right, all the options so far use `_`.

Reply:
Switched to using an underscore. And `StrOptions` in the `_parameter_constraints`.